SETUP/tests/misc_FlattenDirectoryTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+class FlattenDirectoryTest extends PHPUnit\Framework\TestCase
+{
+    const EMPTY_DIRECTORY_PATH = 'data/empty_directory';
+
+    public function testFlatteningEmptyDirectory()
+    {
+        mkdir(self::EMPTY_DIRECTORY_PATH);
+        flatten_directory(self::EMPTY_DIRECTORY_PATH);
+        rmdir(self::EMPTY_DIRECTORY_PATH);
+
+        // PHPUnit labels tests without at least one assert as risky
+        $this->assertTrue(true);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testFlatteningNonExistentDirectory() {
+        flatten_directory('data/non_existent');
+    }
+
+    public function testFlatteningDirectoryWithNestedSubdirectories() {
+        flatten_directory('data/nested_directories');
```

Comment:
Tests should not alter files (or directories) committed into the repo. Instead, create (and clean up) the files/directories within the testcase itself.

Reply:
Good catch, too bad PHP doesn't have a nice recursive delete function. :)